### DISCUSSION TITLE
change fracture criterion in XFEM

### DISCUSSION THREAD

**Discussion by wsfsz123** [idaholab/moose#16186] [Q&A Modules: General] [answered]
Dear all,

I am trying to embed anisotropic fracture toughness property into XFEM, the crack direction is determined by the maximum energy release rate criterion (max(G(θ)/Gc(θ)), have anyone tried this? The current examples in XFEM modules are using maximal principle stress, can anyone help me to change the criterion?

Thanks!

**Comment by jiangwen84** [comment]
Are you planning to use a domain integral to compute the energy release rate G?

**Comment by wsfsz123** [comment, marked as answer]
Yes, the main purpose is to introduce the anisotropic fracture energy (Gc), then compare it to G to predict the crack direction. Thus I need to compute the instant integral firstly.

Thanks,
Shizhe

**Reply by jiangwen84** [reply]
We have published a paper before using K as the criterion for crack propagation. https://www.sciencedirect.com/science/article/pii/S0013794419307568

However, the codes have not merged into the MOOSE yet. We have post processor to compute K/J, and what is missing is to hook the post processor to XFEM crack cut object. 

If we can find some time, I am very interested to add such capability in MOOSE.

**Reply by Wang-Yang-51** [reply]
> 在使用 K 作为裂纹扩展的标准之前，我们已经发表了一篇论文。https://www.sciencedirect.com/science/article/pii/S0013794419307568
> 
> 但是，这些代码尚未合并到 MOOSE 中。我们有后处理器来计算 K/J，而缺少的是将后处理器挂接到 XFEM 裂纹切割对象。
> 
> 如果我们能找到时间，我非常有兴趣在 MOOSE 中添加这样的功能。

Dear Wen Jiang，
I am using a UMAT and UDMGINI subroutine in Abaqus, combined with XFEM, to simulate fatigue crack propagation. The model runs without errors, but the crack does not propagate in the results.
I have tried adjusting analysis steps, boundary conditions, grain crystal orientations, and the damage initiation criterion in the UDMGINI subroutine, but the results are still unsatisfactory. I am unsure how to further modify my model and would appreciate any guidance or assistance.
Alternatively, if you could provide a successful case model with crack propagation for reference, I would be extremely grateful.
[.inp and .for document.zip](https://github.com/user-attachments/files/19544080/default.inp.and.for.document.zip)
![微信截图_20250328095121](https://github.com/user-attachments/assets/fa10b958-fa3d-4d2d-9fa9-930d6401e4ec)
![微信截图_20250328095307](https://github.com/user-attachments/assets/be40a3e5-58d4-41d5-af69-3cbb530fb569)

**Comment by wsfsz123** [comment]
Thank you for your attention. I have read your paper and see the examples on MOOSE. It would be grateful if you could take time to do it, then we can use it with anisotropic fracture energy of material, up to now, there are rare investigations about it, and a good simulation tool with suitable criterion is required.

Thanks,
Shizhe

**Reply by jiangwen84** [reply]
Are you primarily interested in 2D or 3D? single crack or multiple cracks?

**Comment by wsfsz123** [comment]
2D primarily, and a single crack is enough for now, but we are also interested in the dynamic crack in the future. Is it possible to input Gc as a function of theta (the format of anisotropic fracture energy)?

**Reply by Rose-Ama** [reply]
Just a follow-up. Has this been done? Is it possible to create a custom code to solve this problem?
